### PR TITLE
feat(content-manager): add syntax highlighting in the code block type in the Blocks field

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

Adds the highlight colors in the Code blocks inside Block Editor.

### Why is it needed?

We want to highlight the syntax of every single language.
We used Prismjs to do it and this theme https://prismjs.com/index.html?theme=prism-solarizedlight if we want something else just add a comment.

### How to test it?

 User adds a code block in the Blocks field
 - GIVEN THAT I have selected the code block type,
 - WHEN I enter code into the code block
   - THEN the code is displayed with syntax highlighting according to the language detected or specified

 User selects a specific programming language for syntax highlighting
 - GIVEN THAT I am editing a code block in the Blocks field,
 - WHEN I choose a programming language from a dropdown list (e.g., JavaScript, Python, HTML),
   - THEN the code block applies syntax highlighting specific to the selected language

### Related issue(s)/PR(s)

CS-865

### CODE REVIEW

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/Blocks/tests/Code.test.tsx [hunk 1]
```diff
@@ -30,6 +31,7 @@ describe('Code', () => {
 
     expect(screen.getByRole('code')).toBeInTheDocument();
     expect(screen.getByText('Some code')).toBeInTheDocument();
+    screen.debug();
```
Comment (markkaylor) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/Blocks/tests/Code.test.tsx:34]: 👀

Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/Blocks/tests/Code.test.tsx:34]: ouch

File: .github/workflows/adminBundleSize.yml [hunk 2]
```diff
@@ -37,8 +37,8 @@ jobs:
         with:
           build-script: 'build:size'
           pattern: '**/build/**/*.{js,css,html,svg}'
-          strip-hash: "-([-\\w]{8})(\\.\\w+)$"
-          minimum-change-threshold: '5%'
+          strip-hash: "-(\\w{8})(\\.\\w+)$"
+          minimum-change-threshold: 10
```
Comment (remidej) [inline .github/workflows/adminBundleSize.yml:41]: This looks like it should be a dedicated PR

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts [hunk 3]
```diff
@@ -1,214 +1,267 @@
-export const codeLanguages: { value: string; label: string }[] = [
+export const codeLanguages: { value: string; label: string; decorate?: string }[] = [
   {
     value: 'asm',
     label: 'Assembly',
+    decorate: 'asmatmel',
```
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts:5]: yes I think so

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksContent.tsx
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksContent.tsx:1]: I followed these guide in Slate https://github.com/ianstormtaylor/slate/blob/main/site/examples/ts/code-highlighting.tsx as you can see I need to pass the decorate method in the EditableComponent

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (innerdvations) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts:5]: looks like yaml differs too, so it's probably safer to explicitly declare it, even if it's more verbose

**Comment by simotae14** [comment]
In the end I found that downloading and creating a file with the CSS and the js from Prismjs with the necessary languages solved the import issue and let us use just the code we needed for the highlight. I hope this solution can work for you

**Comment by simotae14** [comment]
I am gonna close this because it is a duplication of this PR https://github.com/strapi/strapi/pull/22332
